File: app/src/main/java/io/github/marcocipriani01/telescopetouch/activities/DiagnosticActivityModule.java
/*
 * Copyright 2021 Marco Cipriani (@marcocipriani01)
 *
 *  This program is free software: you can redistribute it and/or modify
 *  it under the terms of the GNU General Public License as published by
 *  the Free Software Foundation, either version 3 of the License, or
 *  (at your option) any later version.
 *
 *  This program is distributed in the hope that it will be useful,
 *  but WITHOUT ANY WARRANTY; without even the implied warranty of
 *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *  GNU General Public License for more details.
 *
 *  You should have received a copy of the GNU General Public License
 *  along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */

package io.github.marcocipriani01.telescopetouch.activities;

import android.app.Activity;
import android.content.Context;
import android.os.Handler;
import android.os.Looper;

import dagger.Module;
import dagger.Provides;
import io.github.marcocipriani01.telescopetouch.inject.PerActivity;

/**
 * @author johntaylor
 */
@Module
public class DiagnosticActivityModule {

    private final DiagnosticActivity activity;

    public DiagnosticActivityModule(DiagnosticActivity activity) {
        this.activity = activity;
    }

    @Provides
    @PerActivity
    Activity provideActivity() {
        return activity;
    }

    @Provides
    @PerActivity
    Context provideActivityContext() {
        return activity;
    }

    @Provides
    @PerActivity
    Handler provideHandler() {
        return new Handler(Looper.getMainLooper());
    }
}

File: app/src/main/java/com/google/android/stardroid/activities/DiagnosticActivityModule.kt
package com.google.android.stardroid.activities

import android.app.Activity
import android.content.Context
import android.os.Handler
import com.google.android.stardroid.inject.PerActivity
import dagger.Module
import dagger.Provides

/**
 * Created by johntaylor on 4/15/16.
 */
@Module
class DiagnosticActivityModule(private val activity: DiagnosticActivity) {
    @Provides
    @PerActivity
    fun provideActivity(): Activity {
        return activity
    }

    @Provides
    @PerActivity
    fun provideActivityContext(): Context {
        return activity
    }

    @Provides
    @PerActivity
    fun provideHandler(): Handler {
        return Handler()
    }
}